internal/pkg/agent/application/upgrade/artifact/download/http/verifier.go

```diff
@@ -106,11 +113,26 @@ func (v *Verifier) Verify(a artifact.Artifact, version string) error {
 	return nil
 }
 
-func (v *Verifier) verifyAsc(a artifact.Artifact, version string) error {
-	if len(v.pgpBytes) == 0 {
+func (v *Verifier) verifyAsc(a artifact.Artifact, version string, pgpBytes ...string) error {
```

Comment:
this looks awfully similar to the fs verifier... Maybe we can separate the mechanism of getting the file from the verification? Maybe we can compose different verifiers depending on the way we are trying to upgrade if we need to do additional checks other than pgp signature ones?